tracing-appender/src/rolling.rs

```diff
@@ -0,0 +1,133 @@
+use crate::inner::InnerAppender;
+use chrono::{DateTime, Datelike, TimeZone, Timelike, Utc};
+use std::fmt::Debug;
+use std::io;
+use std::path::Path;
+
+#[derive(Debug)]
+pub struct RollingFileAppender {
+    inner: InnerAppender,
+}
+
+impl RollingFileAppender {
+    pub fn new(
+        rotation: Rotation,
+        directory: impl AsRef<Path>,
+        file_name_prefix: impl AsRef<Path>,
+    ) -> RollingFileAppender {
+        RollingFileAppender {
+            inner: InnerAppender::new(
+                directory.as_ref(),
+                file_name_prefix.as_ref(),
+                rotation,
+                Utc::now(),
+            )
+            .expect("Failed to create appender"),
+        }
+    }
+}
+
+impl io::Write for RollingFileAppender {
+    fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
+        self.inner.write(buf)
+    }
+
+    fn flush(&mut self) -> io::Result<()> {
+        self.inner.flush()
+    }
+}
+
+/// Creates an hourly, rolling file appender.
+///
+/// The appender returned by `rolling::hourly` can be used with `non_blocking` to create
+/// a non-blocking, hourly file appender.
+///
+/// # Examples
+///
+/// ``` rust
+///
+/// #[clippy::allow(needless_doctest_main)]
+///
```

Comment:
can we add a leading `#` to the clippy attribute, so that it doesn't show up in the rendered docs? Like
```suggestion
/// # #[clippy::allow(needless_doctest_main)]
```